electrum/gui/kivy/main_window.py

```diff
@@ -1213,11 +1218,60 @@ def check_pin_code(self, pin):
         if pin != self.electrum_config.get('pin_code'):
             raise InvalidPassword
 
+    def check_password_for_directory(self, old_password, new_password, update=False):
```

Comment:
Could you either add a docstring or change the API?
E.g. 
```
"""Checks password against all wallets and returns (list of matches, list of mismatches).
If update is True, it also changes the wallet passwords to new_password.
"""
```
Alternatively, the API might be clearer if "update" and "new_password" are merged together. Simplest would be to set `new_password` to `None` to correspond to `update==False`, however `None` is a valid value for `new_password` I guess, so we could introduce a sentinel e.g.
```
SENTINEL=object()
def check_password_for_directory(self, old_password, new_password=SENTINEL): ...
```

Reply:
I agree, it is better to change the API. `None` is not a valid password for Kivy, so I guess it is fine to use it as long as this method stays in the GUI